# HackBio Internship 2024

[Link to the official website](https://thehackbio.com/)

![icon](https://github.com/user-attachments/assets/e5898106-82fb-4b0b-b252-e50282ee5706)

Welcome to my repository for the HackBio Internship 2024, a collaborative and immersive experience to advance bioinformatics skills through real-world data projects.

## About HackBio

HackBio is a bioinformatics internship founded to bridge the gap between knowledge and application in computational biology. With a strong focus on mentorship, collaboration, and hands-on learning, HackBio allowed us, aspiring scientists and bioinformaticians, to work on significant projects with real-life implications.

## Track: AMR in Cancer

My teammates and I worked on the Antimicrobial Resistance (AMR) in Cancer track for this project.

Antimicrobial resistance roughly refers to the ability of microbes (_such as bacteria, fungi, viruses, and parasites_) to evolve and resist the effects of drugs that were once effective against them. This phenomenon is challenging in cancer treatment, particularly for immunocompromised patients undergoing chemotherapy, where infections caused by resistant pathogens can lead to complications.


## Tools and Software Used

Throughout the internship, Python, R, and Bash were used.

---

# Overview

## Stage 0: Introduction to Drug Resistance

**Goal**: Write an essay (400 words) on Drug Resistance in cancer using references. 

![alt text](https://scandiononcology.com/wp-content/uploads/2021/12/Cell-illustration-What-is-cancer-drug-resistance_v5.png)

**Implication**: This step involves understanding the intersection between cancer treatment and drug resistance, where resistance to antibiotics affects treatment outcomes, particularly in cancer patients. For example, it can influence immunotherapy efficiency. This stage focuses on research synthesis and scientific communication, explaining the challenges of managing infections in cancer patients while balancing antibiotic use.

## Stage 1: Literature Review

**Goal**: Analyze the paper "Gut Resistome of NSCLC Patients Treated with Immunotherapy" & make a short video about it.

![image](https://github.com/user-attachments/assets/24945a68-efd7-4d86-81b4-af24f2957c84)

**Implication**: The purpose here is to understand how gut microbiomes contribute to AMR in cancer treatments, particularly for non-small cell lung cancer (NSCLC) patients using immunotherapy. The task helps highlight how AMR genes can reduce the effectiveness of cancer treatments, providing insights into future personalized cancer therapies by managing resistance factors in the gut microbiome. Collaborative team engagement and research presentation are crucial here.

## Stage 2: AMR Data Analysis

**Goal**: Download and clean the AMR Products Dataset, perform statistical analysis, visualize data, and generate key insights about AMR trends. 

**Implication**: The task is aimed at developing data analysis skills by cleaning, processing, and interpreting an AMR dataset. By visualizing trends in AMR product development, it’s possible to identify promising drug candidates and understand AMR dynamics in different regions. This stage focuses on technical skills in data science (e.g., using Python or R), and critical thinking to extract insights that can influence public health strategies.

## Stage 3: AMR Data Analysis and Dashboard Generation

**Goal**: Introduce bioinformatics using Bash scripting for a simulated wet-lab biologist, alongside analyzing a WHO dataset about cholera outbreaks, using R.

**Implication**: This phase provides foundational bioinformatics tasks with an emphasis on Bash scripting for sequencing data analysis and file navigation. By offering simple, yet powerful, scripts, wet-lab biologists can automate tasks like genome assembly, variant calling, and data preprocessing. The parallel analysis of the cholera outbreak dataset exemplifies the importance of bioinformatics tools in handling global health data. Through this, we demonstrate how key insights from large datasets like cholera outbreaks can inform public health decisions and antimicrobial resistance (AMR) strategies.

## Stage 4: NGS Pipeline Development

**Goal**: Build an NGS analysis pipeline using bash or another workflow manager (e.g., Nextflow, Snakemake) for real datasets. 

**Implication**: This final stage integrates various bioinformatics tools (FastQC, FastP, BWA, FreeBayes) into a cohesive pipeline for NGS data analysis. By running tasks like quality control, trimming, genome alignment, and variant calling in an automated fashion, it provides a reproducible framework for researchers to analyze multiple datasets. This step demonstrates technical proficiency in building scalable and efficient pipelines for genomics research.

---

# Collaborators

Shoutout to all my fellow intern-mates who collaborated on this internship!

| Authors | Name | GitHub | LinkedIn |
| :---: | :---: | :---: | :---: |
| 1 | Sarani Nativel-Santerne | https://github.com/Sarani-NS | https://ww.linkedin.com/in/sarani-nativel-santerne |
| 2 | Yetunde Alo Mary | https://github.com/aloyetunde | https://www.linkedin.com/in/yetunde-alo-mary |
| 3 | Nada ElSayed Ahmed | https://github.com/Nada-EA | https://www.linkedin.com/in/nada-elsayed-ahmed |
| 4 | Nada ElHadidy | https://github.com/nada-elhadidy | https://www.linkedin.com/in/nada-elhadidy |
| 5 | Astrid Liliana Vargas | https://github.com/Liliana223 | https://www.linkedin.com/in/astrid-liliana-vargas |
| 6 | Dua Gulzar | https://github.com/duagulzar | http://linkedin.com/in/dua-gulzar |
| 7 | Premalata Pati | https://github.com/premalata23 | https://www.linkedin.com/in/dr-premalata-pati |
| 8 | Hana Mahmoud Mazroua | https://github.com/hanamahmoudd29 | https://www.linkedin.com/in/hana-mazroua |
| 9 | Nourhan Adel ElHifnawy | https://github.com/NourhanElhifnawy | https://www.linkedin.com/in/nourhan-elhifnawy |
| 10 | Jesse Frank Juma | https://github.com/frank1645 | https://www.linkedin.com/in/jesse-frank-juma |


 

 
